### PR TITLE
Add circular median

### PR DESCRIPTION
**Pull request by nstarman**
Found myself wanting this in a recent bit of code.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by nstarman** [comment]
@keflavich, is the doctest sufficient testing or do you want something more?

File: astropy/stats/circstats.py [hunk 1]
```diff
@@ -134,6 +135,45 @@ def circmean(
...
+    >>> data = np.array([51, 67, 40, 109, 31, 405, 358])*u.deg
+    >>> circmedian(data)
+    <Quantity 45. deg>
+
+    References
+    ----------
+    .. [1] https://insidebigdata.com/2021/02/12/circular-statistics-in-python-an-intuitive-intro/
+    """
+    theta = np.arctan2(
+        np.median(np.sin(data), axis=axis),
```
Comment (mhvk) [inline astropy/stats/circstats.py:169]: This feels not quite right, and indeed I could fairly easily find an example where it does not give the right answer:
```
In [11]: a = np.linspace(0, 135, 1001) << u.deg

In [12]: np.arctan2(np.median(np.sin(a)), np.median(np.cos(a))) << u.deg
Out[12]: <Quantity 65.27434311 deg>
```
(should be 67.5, as correctly found by replacing .median with .mean)

Note that ref. [1] also does something quite different.

**Comment by nstarman** [comment]
I've done some more reading and found https://hci.iwr.uni-heidelberg.de/sites/default/files/profiles/mstorath/files/storath2017fast.pdf.
Looking at Fig 2, there actually isn't a good definition of the circular median!
We can more robustly define the circular medoid.

**Comment by larrybradley** [comment]
Also see https://github.com/scipy/scipy/issues/6644

**Comment by larrybradley** [comment]
I think this should go in Astropy. That's scipy issue is 8 years old and they haven't converged.  I was just pointing it out that discussion about multiple definitions of the median.

**Comment by larrybradley** [comment]
What is the median for the case of evenly-distributed points around a circle?

**Comment by mhvk** [comment]
> What is the median for the case of evenly-distributed points around a circle?

Logically, undefined, as for the mean...

**Comment by nstarman** [comment]
Well, I guess that means closing this PR.
https://github.com/astropy/astropy/pull/17502#issuecomment-2520844214 lays out the steps for 2 preceding PRs.

**Pull request closed without merging**